### DISCUSSION TITLE
how to limit the width of ui.label?

### DISCUSSION THREAD

**Discussion by yanghw116** [zauberzeug/nicegui#2497] [Q&A] [answered]
### Question

hi, everyone, I'm new to nicegui.

code example:
```py
from nicegui import ui

with ui.card():
    ui.label('hello')
    ui.label('ABCDEFGHIJKLMNOPQRSTUVWXYZ')

ui.run()
```

here I created 2 `ui.label`. one is short and the other is extremely long. how to set the width of the second `ui.label` to that of the first one and append "..." to the second one to suggest that it's text is too long to be shown here?

**Comment by falkoschindler** [comment, marked as answer]
Hi @yanghw116, for a fixed card width it's not too hard:
```py
with ui.card().classes('w-32 items-stretch'):
    ui.label('hello')
    ui.label('ABCDEFGHIJKLMNOPQRSTUVWXYZ').classes('overflow-hidden text-ellipsis')
```

But for automatically adjusting the width of the card to the first label, I haven't found a solution yet. It's basically a HTML/CSS problem.

**Reply by yanghw116** [reply]
@falkoschindler 
I tried your code, and found something strange.
```
# works fine
ui.label('random_string:ABCDEFGHIJKLMNOPQRSTUVWXYZ').classes('overflow-hidden text-ellipsis')
# line break. only difference is a space inserted right after ':'
ui.label('random_string: ABCDEFGHIJKLMNOPQRSTUVWXYZ').classes('overflow-hidden text-ellipsis')
```
seems that some special characters besides a whiter space will cause line break. How to solve this?

**Reply by falkoschindler** [reply]
@yanghw116 You can add the class "whitespace-nowrap" to prevent the string from wrapping.

**Reply by yanghw116** [reply]
great! it's solved.

**Comment by NicholasQu** [comment]
to fullfill the same width requirement, i believe you just compare the 2 string's length,  and manually cut the long sting to abbr , 
which has the same length with short one. 
It must be easier to implement, while  css solution should be harder.

**Reply by yanghw116** [reply]
this may not be a good solution as the text font size may be different (not in this case, but other cases).
